vmware-event-router/internal/processor/aws_event_bridge_test.go

```diff
@@ -0,0 +1,93 @@
+package processor
+
+import (
+	"testing"
+
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vmware-event-router/internal/metrics"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+func Test_batching_createPutEventsInput(t *testing.T) {
+	tests := []struct {
```

Comment:
can we have a test case with events in the `[]BaseEvent` slice that we don't have a subscription for? E.g. to test a large number of events in the incoming event slice but only a subsection of them will make it into the final batch(es) sent to AWS.

Reply:
Done :+1:

Reply:
Sorry, I probably miscommunicated. What I was asking for is different types of events in the same `[]BaseEvent`. E.g. having `VmPoweredOnEvent` and `VmPoweringOnWithCustomizedDVPortEvent` mixed but a batch entry should only be created for one of these (e.g. pattern map only wants `VmPoweredOnEvent`). This is very possible and should be tested as well.

Reply:
Yeah Michael, this is a totally a valid test case. I added a method which mixes events of two types and sends them to the processor, verifying that only the desired event (to which the processor is subscribed) is put into the batches which are to be sent to the AWS Event Bridge :+1: